sqlmesh/core/state_sync/base.py

```diff
@@ -429,6 +428,28 @@ def migrate(
     def rollback(self) -> None:
         """Rollback to previous backed up state."""
 
+    def add_interval(
+        self,
+        snapshot: Snapshot,
```

Comment:
why don't we just make add_interval accept a snapshot id? the reason for this in the past was probably because intervals were attached to snapshots but this is no longer true

why do we need to split out this method?

Reply:
We discussed this option over a call.